### PR TITLE
Lut generation and schema search

### PR DESCRIPTION
**Pull request by srikanth-iyer**
Uploading partial LUT functions from my branch to a dedicated CASCI branch

### CODE REVIEW

File: cana/boolean_network.py [hunk 1]
```diff
@@ -164,20 +174,30 @@ def from_file(self, file, type="cnet", keep_constants=True, **kwargs):
...
+                    **kwargs,
                 )
             elif type == "logical":
                 return self.from_string_boolean(
                     infile.read(), keep_constants=keep_constants, **kwargs
                 )
 
     @classmethod
-    def from_string_cnet(self, string, keep_constants=True, **kwargs):
+    def from_string_cnet(
```
Comment (jcrozum) [inline cana/boolean_network.py:181]: Move this stuff into the `boolean_node.py` file, and just call the `BooleanNode` constructor to handle clashes and unspecified outputs

File: cana/boolean_node.py [hunk 2]
```diff
@@ -817,7 +835,7 @@ def _check_compute_canalization_variables(self, **kwargs):
             raise Exception("Canalization variable name not found. %s" % kwargs)
         return True
 
-    def bias(self):
+    def bias(self, verbose=True):
```
Comment (jcrozum) [inline cana/boolean_node.py]: default to `False`

File: cana/boolean_node.py [hunk 3]
```diff
@@ -830,7 +848,19 @@ def bias(self):
         See Also:
             :func:`~cana.boolean_network.BooleanNetwork.network_bias`
         """
-        return sum(map(int, self.outputs)) / 2**self.k
+        if verbose:
+            if "?" in self.outputs:
+                print(
```
Comment (jcrozum) [inline cana/boolean_node.py]: make warning

File: tests/test_boolean_node.py [hunk 4]
```diff
@@ -453,18 +740,183 @@ def test_input_symmetry_RULE90():
...
+            generated_node.outputs == expected_output_list
+        ), f"from_partial_lut failed: {generated_node.outputs} != {expected_output_list}"
+
+
+def test_generate_with_required_node_bias():
+    # removing some effective graph inputs to make them incomplete
+    incomplete_automata = {
+        "GKL": [["##0#0#0", 0],["###10#0", 0],["##00###", 0],["0##0###", 0],["###1##1", 1],["1#10###", 1],["1#1#1##", 1], ["1#1###1", 1],],  # ["0###0#0", 0], ["###11##", 1], is missing
+        "GP": [["0#01###", 0],["0##10##", 0],["0#####0", 0],["###0##0", 0],["1#####1", 1],["1##1###", 1],["##10##1", 1],["###01#1", 1],], # ["0#0#00#", 0], is missing
+        "ABK": [["0#01###", 0],["0##10##", 0],["0#####0", 0],["###0##0", 0],["##1#1#1", 1],["1#####1", 1],["1##1###", 1],["##10##1", 1],], # ["0#0#0##", 0],["###01#1", 1], is missing
```
Comment (jcrozum) [inline tests/test_boolean_node.py]: rename GP=ABK

**Review by jcrozum — CHANGES_REQUESTED** [review]

File: cana/boolean_network.py (hunk 1 above)
Comment (srikanth-iyer) [reply cana/boolean_network.py:181]: removed all boolean network changes for now. will incorporate this at a later date after all the existing work is done.

File: tests/test_boolean_node.py (hunk 4 above)
Comment (srikanth-iyer) [reply tests/test_boolean_node.py]: Done

File: cana/boolean_node.py (hunk 3 above)
Comment (srikanth-iyer) [reply cana/boolean_node.py]: done

File: cana/boolean_node.py (hunk 2 above)
Comment (srikanth-iyer) [reply cana/boolean_node.py]: done

**Review by jcrozum — APPROVED** [review]

**Pull request merged**